packages/core/src/internal/clients/leaderboard/InternalLeaderboardClient.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  LeaderboardDelete,
+  LeaderboardFetch,
+  LeaderboardLength,
+  LeaderboardRemoveElements,
+  LeaderboardUpsert,
+} from '../../../messages/responses/leaderboard';
+import {
+  LeaderboardFetchByRankCallOptions,
+  LeaderboardFetchByScoreCallOptions,
+  LeaderboardGetRankCallOptions,
+} from '../../../utils/cache-call-options';
+
+export interface InternalLeaderboardClient {
```

Comment:
I'm least certain about this interface. Should it be more like the [AbstractCacheClient](https://github.com/momentohq/client-sdk-javascript/blob/1c6a68b912734aa92650ac0728df37afffbcbd28/packages/core/src/internal/clients/cache/AbstractCacheClient.ts) class setup for CacheClients?

Reply:
See my comments on the concrete client class, they may be relevant here.

IMO this interface looks like exactly what I was expecting to see for `ILeaderboard`, so I would rename it to that, move it to a public folder, and then make it the return type for the public `.leaderboard()` function.

You can still make the concrete internal client class implement this for now if it makes it easier to get the API into the shape we are expecting for the bug bash.